omegaconf/listconfig.py

```diff
@@ -579,10 +579,7 @@ def _set_value_impl(
             self.__dict__["_content"] = None
         elif vk is ValueKind.MANDATORY_MISSING:
             self.__dict__["_content"] = "???"
-        elif vk in (
-            ValueKind.INTERPOLATION,
-            ValueKind.STR_INTERPOLATION,
-        ):
+        elif vk == ValueKind.INTERPOLATION:
```

Comment:
Is your concern related to the detection of interpolations as strings containing `${`?

If yes I still think it is correct. Such strings can only be 3 things:
* Valid interpolations (ok)
* Invalid interpolations, e.g. `"${"` (an error will be raised when trying to resolve them)
* Escaped interpolations, e.g. `"\${a}"` (they must be parsed as interpolations to be properly resolved -- we can even see them as special kinds of string interpolations)

Reply:
> Invalid interpolations, e.g. "${" (an error will be raised when trying to resolve them)

This is a behavior change:
```
In [16]: OmegaConf.create({"d": ListConfig('${a}')})
Out[16]: {'d': '${a}'}

In [17]: OmegaConf.create({"d": ListConfig('${a')})
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
...
ValidationError: Invalid value assigned : str is not a ListConfig, list or tuple.
```

Reply:
True, the error detection is delayed. I thought it'd be a reasonable trade-off but I can force a parse if you prefer (since it would still be gated by the `${` check it wouldn't slow things down too badly). But note that:
- The error would probably be different
- The previous checks were far from perfect in the first place, for instance on master the following invalid strings are also accepted at creation time: `'${a}}'`, `'{${a}}'`, `${a/b/c}`
Let me know what you prefer

Reply:
Yeah, let's do a real parse on value assignment.
Note: I don't think this should be recursive. just validate that the string looks like a proper interpolation. no need to evaluate nested interpolations and so on.

> The error in your example would probably be different

This is fine.

> The previous checks were far from perfect in the first place, for instance on master the following invalid strings are also accepted at creation time: '${a}}', '{${a}}', ${a/b/c}

Also fine.

Reply:
Done in cd07b48
(NB: also pushed 9306272 which solves annoying crashes when handling error messages containing the $ sign)